src/components/FlowStep/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { FC } from "react"
+import { Box, BoxProps, Icon, useMultiStyleConfig } from "@chakra-ui/react"
+import { BodyMd, BodySm, LabelLg, LabelSm, LabelXs } from "../Typography"
+import { IoArrowForwardCircleSharp } from "react-icons/all"
+import {
+  FlowStepSize,
+  FlowStepStatus,
+  FlowStepVariant,
+} from "../../theme/FlowStep"
+
+export interface FlowStepProps extends BoxProps {
+  status?: FlowStepStatus
+  variant?: FlowStepVariant
+  size?: FlowStepSize
+  preTitle: string
+  title: string
+  isDescriptionHidden?: boolean
+  isDescriptionArrowHidden?: boolean
+}
+
+export const FlowStep: FC<FlowStepProps> = (props) => {
+  const {
+    status = FlowStepStatus.inactive,
+    variant = FlowStepVariant.vertical,
+    size = FlowStepSize.lg,
+    preTitle,
+    title,
+    children,
+    isDescriptionHidden = false,
+    isDescriptionArrowHidden = false,
+    ...rest
+  } = props
+  const styles = useMultiStyleConfig("FlowStep", props)
+
+  const isActive = status === FlowStepStatus.active
+  const isComplete = status === FlowStepStatus.complete
+  const isSmall = size === FlowStepSize.sm
+  const PreTitle = isSmall ? LabelXs : LabelSm
+  const Title = isSmall ? LabelSm : LabelLg
+  const Description = isSmall ? BodySm : BodyMd
+
+  return (
+    <Box __css={styles.wrapper} {...rest}>
+      <Box __css={styles.divider} />
+      <Box __css={styles.subWrapper}>
+        <PreTitle __css={styles.preTitle}>{preTitle}</PreTitle>
+        <Title __css={styles.title}>{title}</Title>
+        {!isDescriptionHidden &&
+          !isComplete &&
+          (typeof children === "string" ? (
+            <Box display="flex">
+              {!isDescriptionArrowHidden && isActive && (
+                <Icon
+                  __css={styles.descriptionArrow}
+                  as={IoArrowForwardCircleSharp}
+                />
+              )}
+              <Description>{children}</Description>
+            </Box>
+          ) : (
+            children
+          ))}
+      </Box>
+    </Box>
+  )
+}
+
+export default FlowStep
```

Comment:
We should export the FlowStep component in `components/index.ts` file.

`export * from "./FlowStep"`

This will export FlowStep components and also types and enums if you move them here (see comment above).

I think we can get rid of `export default FlowStep` here since we will have to export and import all the things from this file anyway.